lib/puppet/pops/adapters.rb

```diff
@@ -128,7 +132,7 @@ def self.find_module_for_dir(environment, paths, dir)
         end
         if relative_path.length > 1
           mod = environment.module(relative_path[0])
-          return mod unless mod.nil?
```

Comment:
why remove this?

Reply:
Ugh... I can't, I don't think. I was walking through why I could, but I think this change will cause an error if users have nested modulepaths, eg "/etc/puppetlabs/code/modules:/etc/puppetlabs/code/modules/mysupermodules/"